ChocolateyGet.psm1

```diff
@@ -139,7 +174,42 @@ function Find-Package {
         Write-Error ( $LocalizedData.SearchingEntireRepo)
         return
     }
-    
+
+    [array]$RegisteredPackageSources = Get-PackageSources    
+        
+    if($options -and $options.ContainsKey('Source'))
+    {        
+        # Finding the matched package sources from the registered ones
+        $sourceName = $options['Source']
+        Write-Verbose ($LocalizedData.SpecifiedSourceName -f ($sourceName))        
+        
+        if($RegisteredPackageSources.Name -eq $sourceName)
+        {
+            # Found the matched registered source
+            $selectedSource = $sourceName              
+        }
+        else
+        {
+            $message = $LocalizedData.PackageSourceNotFound -f ($sourceName)
+            ThrowError -ExceptionName "System.ArgumentException" `
+                -ExceptionMessage $message `
+                -ErrorId "PackageSourceNotFound" `
+                -CallerPSCmdlet $PSCmdlet `
+                -ErrorCategory InvalidArgument `
+                -ExceptionObject $sourceName
+        }
+    }
+    else {
+        $selectedSource = $script:PackageSourceName
```

Comment:
$script:PackageSource?

Reply:
Nope, PackageSourceName. Chocolatey's default package source name is 'chocolatey', so if no other source name is specified, it will try to use the default public repository name (unless someone has removed it from choco's configuration).

TBH, $script:PackageSource as a URL could be removed since it doesn't get used anywhere and is kind of redundant.